SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialVirtualWorkerTranslator.cpp

```diff
@@ -116,6 +114,18 @@ void SpatialVirtualWorkerTranslator::ApplyMappingFromSchema(Schema_Object* Objec
 
 		// Insert each into the provided map.
 		VirtualToPhysicalWorkerMapping.Add(VirtualWorkerId, PhysicalWorkerName);
+		if (PhysicalWorkerName == WorkerId)
+		{
+			LocalVirtualWorkerId = VirtualWorkerId;
+		}
+	}
+
+	if (LocalVirtualWorkerId != SpatialConstants::INVALID_COMPONENT_ID)
+	{
+		bIsReady = true;
```

Comment:
Is  this ready necessary when ConstructVirtualWorkerMappingFromQueryResponse sets ready one it has done it thing?

Reply:
Ahh yes it is. I didn't realize this was inside ApplyMappingFromSchema.

Reply:
Yup, this was one of the bugs we ran into - the Translator has completely different flows depending on if it's authoritative or not.